pkg/core/state/notification_event.go

```diff
@@ -20,6 +20,79 @@ type NotificationEvent struct {
 	Item       *stackitem.Array `json:"state"`
 }
 
+// ContractInvocation contains method call information.
+// The Arguments field will be nil if serialization of the arguments exceeds a predefined limit
+// (for security reasons). In that case Truncated will be set to true.
+type ContractInvocation struct {
+	Hash           util.Uint160     `json:"contracthash"`
+	Method         string           `json:"method"`
+	Arguments      *stackitem.Array `json:"arguments"`
+	ArgumentsBytes []byte
```

Comment:
It's a dangerous pattern. Sometimes only `ArgumentsBytes` is set, and sometimes `Arguments` is set. Whereas in fact at the interop context level you only use `ArgumentsBytes`.It's the RPC server and RPC client who uses `Arguments`.  Ideal solution would be to split core-side and RPC-side Invocation structure into two, but it will lead to the fact that the whole RPC-side `result.ApplicationLog`  has to be rewritten. We can't do that, because earlier we intentionally made this structure reuse as much from the `state` package as possible. Hence, I suggest:
1. Make `ArgumentsBytes` private, document Arguments/argumentsBytes differences properly.
2. Create a constructor for ContractInvocation that accepts arguments bytes.
3. Move arguments decoding to the JSON serialization/deserialization level. Don't decode arguments at the binary encoding level, DB is OK with encoded `argumentsBytes` representation.